[BUG] Compile failure with new MLTask Constructor

### What is the bug?

https://github.com/opensearch-project/ml-commons/pull/2825 added a new parameter to the constructor.  We use that in a test:
> src/test/java/org/opensearch/flowframework/workflow/RegisterLocalPretrainedModelStepTests.java:140: error: no suitable constructor found for MLTask(String,String,,,MLTaskState,,,,,,,,,boolean)

### How can one reproduce the bug?

Attempt to compile the code with current upstream dependencies.

Specifically:
```bash
./gradlew compileTestJava
```
(If it passes, you probably have cached dependencies.)

### What is the expected behavior?

Code compiles.

### What is your host/environment?

Multiple environments failed in https://github.com/opensearch-project/flow-framework/issues/856 but you can probably reproduce it anywhere.

### Do you have any additional context?

The constructor is annotated `@Builder` so we should change that line of code to use a builder (omitting the nulls) to be robust to future changes.

This needs to be fixed promptly and backported to 2.17.

--- a/src/test/java/org/opensearch/flowframework/workflow/DeployModelStepTests.java
+++ b/src/test/java/org/opensearch/flowframework/workflow/DeployModelStepTests.java
@@ -127,22 +127,7 @@ public class DeployModelStepTests extends OpenSearchTestCase {
         // Stub getTask for success case
         doAnswer(invocation -> {
             ActionListener<MLTask> actionListener = invocation.getArgument(1);
-            MLTask output = new MLTask(
-                taskId,
-                modelId,
-                null,
-                null,
-                MLTaskState.COMPLETED,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                false
-            );
+            MLTask output = MLTask.builder().taskId(taskId).modelId(modelId).state(MLTaskState.COMPLETED).async(false).build();
             actionListener.onResponse(output);
             return null;
         }).when(machineLearningNodeClient).getTask(any(), any());
@@ -215,22 +200,7 @@ public class DeployModelStepTests extends OpenSearchTestCase {
         // Stub getTask for success case
         doAnswer(invocation -> {
             ActionListener<MLTask> actionListener = invocation.getArgument(1);
-            MLTask output = new MLTask(
-                taskId,
-                modelId,
-                null,
-                null,
-                MLTaskState.FAILED,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                "error",
-                null,
-                false
-            );
+            MLTask output = MLTask.builder().taskId(taskId).modelId(modelId).state(MLTaskState.COMPLETED).async(false).build();
             actionListener.onResponse(output);
             return null;
         }).when(machineLearningNodeClient).getTask(any(), any());

--- a/src/test/java/org/opensearch/flowframework/workflow/DeployModelStepTests.java
+++ b/src/test/java/org/opensearch/flowframework/workflow/DeployModelStepTests.java
@@ -200,7 +200,7 @@ public class DeployModelStepTests extends OpenSearchTestCase {
         // Stub getTask for success case
         doAnswer(invocation -> {
             ActionListener<MLTask> actionListener = invocation.getArgument(1);
-            MLTask output = MLTask.builder().taskId(taskId).modelId(modelId).state(MLTaskState.COMPLETED).async(false).build();
+            MLTask output = MLTask.builder().taskId(taskId).modelId(modelId).state(MLTaskState.FAILED).async(false).error("error").build();
             actionListener.onResponse(output);
             return null;
         }).when(machineLearningNodeClient).getTask(any(), any());

--- a/src/test/java/org/opensearch/flowframework/workflow/RegisterLocalCustomModelStepTests.java
+++ b/src/test/java/org/opensearch/flowframework/workflow/RegisterLocalCustomModelStepTests.java
@@ -144,22 +144,7 @@ public class RegisterLocalCustomModelStepTests extends OpenSearchTestCase {
         // Stub getTask for success case
         doAnswer(invocation -> {
             ActionListener<MLTask> actionListener = invocation.getArgument(1);
-            MLTask output = new MLTask(
-                taskId,
-                modelId,
-                null,
-                null,
-                MLTaskState.COMPLETED,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                false
-            );
+            MLTask output = MLTask.builder().taskId(taskId).modelId(modelId).state(MLTaskState.COMPLETED).async(false).build();
             actionListener.onResponse(output);
             return null;
         }).when(machineLearningNodeClient).getTask(any(), any());
@@ -239,22 +224,7 @@ public class RegisterLocalCustomModelStepTests extends OpenSearchTestCase {
         // Stub getTask for success case
         doAnswer(invocation -> {
             ActionListener<MLTask> actionListener = invocation.getArgument(1);
-            MLTask output = new MLTask(
-                taskId,
-                modelId,
-                null,
-                null,
-                MLTaskState.COMPLETED,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                false
-            );
+            MLTask output = MLTask.builder().taskId(taskId).modelId(modelId).state(MLTaskState.COMPLETED).async(false).build();
             actionListener.onResponse(output);
             return null;
         }).when(machineLearningNodeClient).getTask(any(), any());
@@ -342,22 +312,7 @@ public class RegisterLocalCustomModelStepTests extends OpenSearchTestCase {
         // Stub get ml task for failure case
         doAnswer(invocation -> {
             ActionListener<MLTask> actionListener = invocation.getArgument(1);
-            MLTask output = new MLTask(
-                taskId,
-                modelId,
-                null,
-                null,
-                MLTaskState.FAILED,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                testErrorMessage,
-                null,
-                false
-            );
+            MLTask output = MLTask.builder().taskId(taskId).modelId(modelId).state(MLTaskState.COMPLETED).async(false).build();
             actionListener.onResponse(output);
             return null;
         }).when(machineLearningNodeClient).getTask(any(), any());

--- a/src/test/java/org/opensearch/flowframework/workflow/RegisterLocalCustomModelStepTests.java
+++ b/src/test/java/org/opensearch/flowframework/workflow/RegisterLocalCustomModelStepTests.java
@@ -312,7 +312,13 @@ public class RegisterLocalCustomModelStepTests extends OpenSearchTestCase {
         // Stub get ml task for failure case
         doAnswer(invocation -> {
             ActionListener<MLTask> actionListener = invocation.getArgument(1);
-            MLTask output = MLTask.builder().taskId(taskId).modelId(modelId).state(MLTaskState.COMPLETED).async(false).build();
+            MLTask output = MLTask.builder()
+                .taskId(taskId)
+                .modelId(modelId)
+                .state(MLTaskState.FAILED)
+                .error(testErrorMessage)
+                .async(false)
+                .build();
             actionListener.onResponse(output);
             return null;
         }).when(machineLearningNodeClient).getTask(any(), any());

--- a/src/test/java/org/opensearch/flowframework/workflow/RegisterLocalPretrainedModelStepTests.java
+++ b/src/test/java/org/opensearch/flowframework/workflow/RegisterLocalPretrainedModelStepTests.java
@@ -232,7 +232,13 @@ public class RegisterLocalPretrainedModelStepTests extends OpenSearchTestCase {
         // Stub get ml task for failure case
         doAnswer(invocation -> {
             ActionListener<MLTask> actionListener = invocation.getArgument(1);
-            MLTask output = MLTask.builder().taskId(taskId).modelId(modelId).state(MLTaskState.COMPLETED).async(false).build();
+            MLTask output = MLTask.builder()
+                .taskId(taskId)
+                .modelId(modelId)
+                .state(MLTaskState.FAILED)
+                .error(testErrorMessage)
+                .async(false)
+                .build();
             actionListener.onResponse(output);
             return null;
         }).when(machineLearningNodeClient).getTask(any(), any());

--- a/src/test/java/org/opensearch/flowframework/workflow/RegisterLocalPretrainedModelStepTests.java
+++ b/src/test/java/org/opensearch/flowframework/workflow/RegisterLocalPretrainedModelStepTests.java
@@ -137,22 +137,7 @@ public class RegisterLocalPretrainedModelStepTests extends OpenSearchTestCase {
         // Stub getTask for success case
         doAnswer(invocation -> {
             ActionListener<MLTask> actionListener = invocation.getArgument(1);
-            MLTask output = new MLTask(
-                taskId,
-                modelId,
-                null,
-                null,
-                MLTaskState.COMPLETED,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                false
-            );
+            MLTask output = MLTask.builder().taskId(taskId).modelId(modelId).state(MLTaskState.COMPLETED).async(false).build();
             actionListener.onResponse(output);
             return null;
         }).when(machineLearningNodeClient).getTask(any(), any());
@@ -247,22 +232,7 @@ public class RegisterLocalPretrainedModelStepTests extends OpenSearchTestCase {
         // Stub get ml task for failure case
         doAnswer(invocation -> {
             ActionListener<MLTask> actionListener = invocation.getArgument(1);
-            MLTask output = new MLTask(
-                taskId,
-                modelId,
-                null,
-                null,
-                MLTaskState.FAILED,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                testErrorMessage,
-                null,
-                false
-            );
+            MLTask output = MLTask.builder().taskId(taskId).modelId(modelId).state(MLTaskState.COMPLETED).async(false).build();
             actionListener.onResponse(output);
             return null;
         }).when(machineLearningNodeClient).getTask(any(), any());

--- a/src/test/java/org/opensearch/flowframework/workflow/RegisterLocalSparseEncodingModelStepTests.java
+++ b/src/test/java/org/opensearch/flowframework/workflow/RegisterLocalSparseEncodingModelStepTests.java
@@ -237,7 +237,13 @@ public class RegisterLocalSparseEncodingModelStepTests extends OpenSearchTestCas
         // Stub get ml task for failure case
         doAnswer(invocation -> {
             ActionListener<MLTask> actionListener = invocation.getArgument(1);
-            MLTask output = MLTask.builder().taskId(taskId).modelId(modelId).state(MLTaskState.COMPLETED).async(false).build();
+            MLTask output = MLTask.builder()
+                .taskId(taskId)
+                .modelId(modelId)
+                .state(MLTaskState.FAILED)
+                .error(testErrorMessage)
+                .async(false)
+                .build();
             actionListener.onResponse(output);
             return null;
         }).when(machineLearningNodeClient).getTask(any(), any());

--- a/src/test/java/org/opensearch/flowframework/workflow/RegisterLocalSparseEncodingModelStepTests.java
+++ b/src/test/java/org/opensearch/flowframework/workflow/RegisterLocalSparseEncodingModelStepTests.java
@@ -140,22 +140,7 @@ public class RegisterLocalSparseEncodingModelStepTests extends OpenSearchTestCas
         // Stub getTask for success case
         doAnswer(invocation -> {
             ActionListener<MLTask> actionListener = invocation.getArgument(1);
-            MLTask output = new MLTask(
-                taskId,
-                modelId,
-                null,
-                null,
-                MLTaskState.COMPLETED,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                false
-            );
+            MLTask output = MLTask.builder().taskId(taskId).modelId(modelId).state(MLTaskState.COMPLETED).async(false).build();
             actionListener.onResponse(output);
             return null;
         }).when(machineLearningNodeClient).getTask(any(), any());
@@ -252,22 +237,7 @@ public class RegisterLocalSparseEncodingModelStepTests extends OpenSearchTestCas
         // Stub get ml task for failure case
         doAnswer(invocation -> {
             ActionListener<MLTask> actionListener = invocation.getArgument(1);
-            MLTask output = new MLTask(
-                taskId,
-                modelId,
-                null,
-                null,
-                MLTaskState.FAILED,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                testErrorMessage,
-                null,
-                false
-            );
+            MLTask output = MLTask.builder().taskId(taskId).modelId(modelId).state(MLTaskState.COMPLETED).async(false).build();
             actionListener.onResponse(output);
             return null;
         }).when(machineLearningNodeClient).getTask(any(), any());

